dpti/workflows/service/file_handler.py

```
@@ -0,0 +1,276 @@
+            target_linkfile_path, start=self.flow_running_dir
+        )
+        return rel_target_linkfile_path
+
+    @staticmethod
+    def create_path_with_backup(path: str) -> str:
+        path += "/"
+        if os.path.isdir(path):
+            dirname = os.path.dirname(path)
+            counter = 0
+            while True:
+                bk_dirname = dirname + ".bk%03d" % counter
```

Ruff (0.11.9)

Use format specifiers instead of percent format

(UP031)